src/marks/contour.d.ts

```diff
@@ -4,15 +4,109 @@ import type {Data, RenderableMark} from "../mark.js";
+   * create an isoline every 15 meters on a topographic map:
+   *
+   * ```js
+   * Plot.contour(volcano.values, {interval: 15, width: volcano.width, height: volcano.height, value: Plot.identity})
+   * ```
+   */
   interval?: RangeInterval;
 }
 
+/**
+ * Return a contour mark, which creates contour polygons from spatial samples.
+ * If *data* is provided, it represents discrete samples in abstract coordinates
+ * *x* and *y*; the *value* channel specifies further abstract values (_e.g._,
+ * height in a topographic map) to be spatially interpolated to produce a raster
+ * grid of quantitative values (like in the **raster** mark), and lastly
+ * contours via marching squares, which are rendered as vector polygons.
+ *
+ * For example, to generate filled contours from a topographic map, where the
+ * color corresponds to the contour threshold value:
+ *
+ * ```js
+ * Plot.contour(volcano.values, {width: volcano.width, height: volcano.height, value: Plot.identity, fill: "value"})
+ * ```
+ *
+ * For smoother contours, the **blur** option (default 0) specifies a
+ * non-negative pixel radius for smoothing prior to applying marching squares.
+ * The **smooth** option (default true) specifies whether to apply linear
+ * interpolation after marching squares when computing contour polygons. The
+ * **thresholds** and **interval** options specify the contour thresholds; see
+ * the **bin** transform for details.
+ *
+ * With the exception of the *x*, *y*, *x1*, *y1*, *x2*, *y2*, and *value*
+ * channels, the contour mark’s channels are not evaluated on the initial *data*
+ * but rather on the contour multipolygons generated in the initializer, as in
+ * the example above.
+ *
+ * The **fill** and **fillOpacity** channels may alternatively be specified as
+ * functions *f*(*x*, *y*) to be evaluated at each pixel centroid of the
+ * underlying raster grid (without interpolation). For example, to draw a
+ * contour plot of a two-dimensional function:
+ *
+ * ```js
+ * Plot.contour({
+ *   fill: (x, y) => Math.sin(x) * Math.cos(y),
+ *   x1: 0,
+ *   y1: 0,
+ *   x2: 4 * Math.PI,
+ *   y2: 4 * Math.PI
+ *  })
+ * ```
+ *
+ * The resolution of the underlying raster grid may be specified with the
+ * following options:
+ *
+ * * **width** - the number of pixels on each horizontal line
+ * * **height** - the number of lines; a positive integer
+ *
+ * The raster dimensions may also be imputed from the extent of *x* and *y* and
+ * a pixel size:
+ *
+ * * **x1** - the starting horizontal position; bound to the *x* scale
+ * * **x2** - the ending horizontal position; bound to the *x* scale
+ * * **y1** - the starting vertical position; bound to the *y* scale
+ * * **y2** - the ending vertical position; bound to the *y* scale
+ * * **pixelSize** - the screen size of a raster pixel; defaults to 1
+ */
 export function contour(options?: ContourOptions): Contour;
-
 export function contour(data?: Data, options?: ContourOptions): Contour;
```

Comment:
In other places I swapped the order of overloaded methods to put the more explicit version first. I think we should probably do that here (and anywhere else), too: it favors the more explicit form when you type `contour(|)` so that you get more explicit guidance about what arguments are expected.